cfgen/cfgen

```diff
@@ -1261,6 +1270,31 @@ class ConfPool(ToDhall):
                           f'pvers = {oids_to_dhall(self.pvers)}'])
         return '{ %s }' % args
 
+    @classmethod
+    def gen_sns_tolerance(cls, m0conf: Dict[Oid, Any],
+                          pool_desc: Dict[str, Any]) -> Failures:
+        encls_nr = len(cluster_encls(m0conf))
+        ctrls_per_encl = floor(len(cluster_ctrls(m0conf)) / encls_nr)
+        data_units = pool_desc['data_units']
+        parity_units = pool_desc['parity_units']
+        spare_units = pool_desc.get('spare_units')
```

Comment:
Potentially, spare_units can be None. On the next line you're summing numbers which is dangerous:

```
Python 3.6.8 (default, Nov 16 2020, 16:55:22)
[GCC 4.8.5 20150623 (Red Hat 4.8.5-44)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> t = None
>>> 2 + t
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unsupported operand type(s) for +: 'int' and 'NoneType'
>>> t + 2
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unsupported operand type(s) for +: 'NoneType' and 'int'
```

```suggestion
        spare_units = pool_desc.get('spare_units', 0)
```

Reply:
Right, thanks.